src/audio-worklet/basic/one-pole-filter/main.js

```diff
@@ -19,15 +21,45 @@ const startAudio = async (context) => {
       .exponentialRampToValueAtTime(0.01, 8.0);
 };
 
+const stopAudio = () => {
+  if (oscillator) {
+    oscillator.stop();
+    oscillator.disconnect();
+  }
+};
+
+
 // A simplem onLoad handler. It also handles user gesture to unlock the audio
 // playback.
 window.addEventListener('load', async () => {
   const buttonEl = document.getElementById('button-start');
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    await startAudio(audioContext);
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
-  }, false);
+    if (!isPlaying) { 
+      await startAudio(audioContext);
```

Comment:
Ditto here.

Reply:
added the flag check here same as above.